### PR TITLE
Add user model with JWT authentication

### PR DESCRIPTION
**Pull request by Dhamsoft**
### Registration: POST /users

```
curl --header "Content-Type: application/json" \
  --request POST \
  --data '{ "user": { "name": "Dhams Nagarajan", "email": "dhams@dhams.com", "password": "123456789", "password_confirmation": "123456789", "contact_number": "35345", "home_address": "86CQ WSellington" } }' \
  http://localhost:3000/users
```

Response => {"name":"Dhams Nagarajan","email":"dhams@dhams.com"}%

### Login: POST users/sign_in

```
curl -i --header "Content-Type: application/json" \
  --request POST \
  --data '{"user": {"email":"dhams@dhams.com","password":"123456789"}}' \
  http://localhost:3000/users/sign_in

```
HTTP/1.1 201 Created
X-Frame-Options: SAMEORIGIN
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
X-Download-Options: noopen
X-Permitted-Cross-Domain-Policies: none
Referrer-Policy: strict-origin-when-cross-origin
Location: /
Content-Type: application/json; charset=utf-8
Authorization: Bearer eyJhbGciOiJIUzI1NiJ9.eyJqdGkiOiJkNDg2MWQ3ZC0yMjI4LTQzNjgtYmE1NC1jZDI5NmE4MTJmMDgiLCJzdWIiOiIyIiwic2NwIjoidXNlciIsImF1ZCI6bnVsbCwiaWF0IjoxNTM5OTEyOTkyLCJleHAiOjE1Mzk5MTY1OTJ9.E6NEu5jZMi_xVM280iI30QOfoZZVU9e2M0k30PNQ5q8
ETag: W/"f1a11abe5791347d260942473ed3662b"
Cache-Control: max-age=0, private, must-revalidate
X-Request-Id: 6f964fc7-6371-48fd-a7cd-f669c222df31
X-Runtime: 0.137653
Transfer-Encoding: chunked

Response => {"name":"Dhams Nagarajan","email":"dhams@dhams.com"}%

### Dashboard: GET /dashboard

curl -i -H "Content-Type: application/json" -H "Authorization: Bearer eyJhbGciOiJIUzI1NiJ9.eyJqdGkiOiJkNDg2MWQ3ZC0yMjI4LTQzNjgtYmE1NC1jZDI5NmE4MTJmMDgiLCJzdWIiOiIyIiwic2NwIjoidXNlciIsImF1ZCI6bnVsbCwiaWF0IjoxNTM5OTEyOTkyLCJleHAiOjE1Mzk5MTY1OTJ9.E6NEu5jZMi_xVM280iI30QOfoZZVU9e2M0k30PNQ5q8"\
  http://localhost:3000/dashboard

HTTP/1.1 200 OK
X-Frame-Options: SAMEORIGIN
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
X-Download-Options: noopen
X-Permitted-Cross-Domain-Policies: none
Referrer-Policy: strict-origin-when-cross-origin
Content-Type: application/json; charset=utf-8
ETag: W/"e1d96d7560c4ac22e3ceb2473e2046ac"
Cache-Control: max-age=0, private, must-revalidate
X-Request-Id: 63deb3d6-e75f-4043-b957-42bb2652ba4e
X-Runtime: 0.009501
Transfer-Encoding: chunked

{"stats":["some stats"]}%

Logout: DELETE users/sign_out
```

curl -i -H "Content-Type: application/json" -H "Authorization: Bearer eyJhbGciOiJIUzI1NiJ9.eyJqdGkiOiJkNDg2MWQ3ZC0yMjI4LTQzNjgtYmE1NC1jZDI5NmE4MTJmMDgiLCJzdWIiOiIyIiwic2NwIjoidXNlciIsImF1ZCI6bnVsbCwiaWF0IjoxNTM5OTEyOTkyLCJleHAiOjE1Mzk5MTY1OTJ9.E6NEu5jZMi_xVM280iI30QOfoZZVU9e2M0k30PNQ5q8"\
    --request DELETE \
  http://localhost:3000/users/sign_out
```

HTTP/1.1 204 No Content
X-Frame-Options: SAMEORIGIN
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
X-Download-Options: noopen
X-Permitted-Cross-Domain-Policies: none
Referrer-Policy: strict-origin-when-cross-origin
Cache-Control: no-cache
X-Request-Id: b89718c3-2b16-4090-a564-f81f9906a6ce
X-Runtime: 0.004705

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class User < ApplicationRecord
+  include Devise::JWT::RevocationStrategies::JTIMatcher
+
+  devise :database_authenticatable, :registerable, :validatable,
+         :recoverable, :jwt_authenticatable, jwt_revocation_strategy: self
+
+  validates :name, presence: true
+
+  def to_json(arg)
```
Comment (Dhamsoft) [inline app/models/user.rb]: TODO: Update with proper serializer

**Comment by kwokster10** [comment]
Testing now.

**Comment by kwokster10** [comment]
Let's split the API documentation into it's own markdown file and make the response clearer. Other than that, looks good.

**Comment by kwokster10** [comment]
LGTM

**Pull request merged**